src/components/user/UserProfile.js

```diff
@@ -9,13 +9,48 @@ import {
   Button,
 } from '@mui/material';
 
-import { Avatar } from '@mui/material';
+import { Avatar, Alert } from '@mui/material';
 import { Link } from 'react-router-dom';
 import { useAuth } from '@akosasante/react-auth-context';
 
 export default function PersonalProfile() {
+  const [purchasedAlbums, setPurchasedAlbums] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
   const { user } = useAuth();
 
+  const fetchPurchasedAlbums = useCallback(async () => {
+    try {
+      const response = await axiosInstance.get(`/users/showMe/withAlbums`);
+      console.log('this is what we get', response.data.user.purchasedAlbums);
+      setPurchasedAlbums(response.data.user.purchasedAlbums);
+      setLoading(false);
+    } catch {
+      console.error('Error fetching purchased albums:', error);
+      setError('Error fetching albums for this user. Please try again later.');
+      setLoading(false);
+    }
+  }, [user.userId]);
+
+  useEffect(() => {
+    fetchPurchasedAlbums();
+  }, [fetchPurchasedAlbums]);
+
+  // useEffect(() => {
+  //   // Fetch user with purchased albums when the component mounts
+  //   const fetchPurchasedAlbums = async () => {
+  //     try {
+  //       const response = await fetch('/api/showMe/withAlbums');
+  //       const data = await response.json();
+  //       setPurchasedAlbums(data.user.purchasedAlbums);
+  //     } catch (error) {
+  //       console.error('Error fetching purchased albums:', error);
+  //     }
+  //   };
+
+  //   fetchPurchasedAlbums();
+  // }, []);
```

Comment:
We could write it this way with all the callback code in a useEffect instead of a separate hook. That would be totally fine as well.

The main things to look out for with what you've commented out is making sure we've set the error and loading state variables, and we'd want to use `axiosInstance` since that one already has our base url set up to work in both local dev and in the deployed code, and is already passing along the authentication cookie. We could do the same with `fetch` but it's unnecessary extra work at this point in the application.